### PR TITLE
[docs] Change log for v0.12.0

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**
\cc @DevExpress/testcafe-docs 

Two bugs are still to come. In the meantime, please review the rest.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,282 @@
...
+The state of webpage elements can now be extended with custom properties.
+
+We have added the [addCustomDOMProperties](https://devexpress,github.io/testcafe/documentation/test-api/selecting-page-elements/selectors.html#adding-custom-properties-to-element-state)
+method to the selector, so that you can add properties to the element state like in the following example.
+
+```js
+import {Selector} from 'testcafe'
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`
```
Comment (AlexanderMoskovkin) [inline CHANGELOG.md]: missed `;`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,282 @@
...
+
+```js
+fixture `My fixture`
+    .page `http://example.com`
+    .httpAuth({
+        username: 'username',
+        password: 'Pa$$word',
+
+        // Optional parameters, can be required for the NTLM authentication.
+        domain: 'CORP-DOMAIN',
```
Comment (kirovboris) [inline CHANGELOG.md]: Incorrect indentation, should be
```javascript
domain:      'CORP-DOMAIN',
workstation: 'machine-win10'
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,282 @@
...
+
+#### Screenshot and window resize actions now work on Linux ([#1117](https://github.com/DevExpress/testcafe/issues/1117))
+
+The `t.takeScreenshot`, `t.resizeWindow`, `t.resizeWindowToFitDevice` and `t.maximizeWindow` actions can now be executed on Linux machines.
+
+#### Adding custom properties to the element state ([#749](https://github.com/DevExpress/testcafe/issues/749))
+
+The state of webpage elements can now be extended with custom properties.
+
+We have added the [addCustomDOMProperties](https://devexpress,github.io/testcafe/documentation/test-api/selecting-page-elements/selectors.html#adding-custom-properties-to-element-state)
```
Comment (helen-dikareva) [inline CHANGELOG.md]: `https://devexpress,github.io/...` --> `https://devexpress.github.io/...`

Comment (helen-dikareva) [inline CHANGELOG.md]: I think we should give links for corresponding documentation topics for other section (e.g. HTTP authentication support)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,282 @@
...
+method to the selector, so that you can add properties to the element state like in the following example.
+
+```js
+import {Selector} from 'testcafe'
```
Comment (helen-dikareva) [inline CHANGELOG.md]: For our repo we use lint rule for this `{Selector}` --> `{ Selector }`. May be it's important for examples too.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,282 @@
...
+ });
+
+ test
+     .page `http://devexpress.github.io/testcafe/blog/`
+     ('Test2', async t => {
+         // Starts at http://devexpress.github.io/testcafe/blog/
+     });
+ ```
+
+#### Initialization and finalization methods for a test ([#1108](https://github.com/DevExpress/testcafe/issues/1108))
```
Comment (helen-dikareva) [inline CHANGELOG.md]: Should we say about overriding `fixture.beforeEach`, `fixture.afterEach` here?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,282 @@
...
+    // originNode === the <ul> element
+});
+```
+
+In addition, all these methods now allow you to pass objects to the predicate's scope on the client. To this end, we have added
+an optional `dependencies` parameter.
+
+```js
+const isNodeOk = ClientFunction(node => { /*...*/ });
+const aFlag = getFlag();
```
Comment (helen-dikareva) [inline CHANGELOG.md]: `aFlag`?)

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,282 @@
...
+
+Selector('ul').child(node => {
+    return isNodeOk(node) && aFlag;
+}, { isNodeOk, aFlag });
+```
+
+#### Filtering by negative index in selectors ([#738](https://github.com/DevExpress/testcafe/issues/738))
```
Comment (helen-dikareva) [inline CHANGELOG.md]: May be better to specify list of methods which take negative index (e.g. `parent` method works another way)

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,282 @@
...
+    .page `http://www.example.com`;
+
+test('My Test', asynct t => {
+    await t.expect(windowLocation).to.eql('http://www.example.com');
+});
+```
+
+#### Automatic waiting for scripts ([#1072](https://github.com/DevExpress/testcafe/issues/1072))
+
+TestCafe now automatically waits for scripts to finish before proceeding to the next test action.
```
Comment (helen-dikareva) [inline CHANGELOG.md]: I think it's unclear

Comment (AlexanderMoskovkin) [inline CHANGELOG.md]: Yep, we'll discuss it in private

File: CHANGELOG.md (hunk 5 above)
Comment (VasilyStrelyaev) [inline CHANGELOG.md]: A link to documentation will be enough

File: CHANGELOG.md (hunk 7 above)
Comment (VasilyStrelyaev) [inline CHANGELOG.md]: what is it about `parent`? you mean it doesn't accept negative index?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,295 @@
...
+* Key identifiers for all keys are now passed to key events ([#1079](https://github.com/DevExpress/testcafe/issues/1079))
+* Touch events are no longer emulated for touch monitors ([#984](https://github.com/DevExpress/testcafe/issues/984))
+* v8 flags can now be passed to Node.js when using TestCafe from the command line ([#1006](https://github.com/DevExpress/testcafe/issues/1006))
+* ShadowUI root is now hidden for `elementFromPoint` in an iframe in IE ([#1029](https://github.com/DevExpress/testcafe/issues/1029))
+* Preventing the form submit event no longer leads to additional delay between actions ([#1115](https://github.com/DevExpress/testcafe/issues/1115))
+* TestCafe no longer hangs when a cursor is moved out of a reloading iframe ([#1140](https://github.com/DevExpress/testcafe/issues/1140))
+* Onclick event handler is now executed correctly during click automation in specific cases ([#1138](https://github.com/DevExpress/testcafe/issues/1138))
+* The `application/pdf` mime type is no longer recognized as a page ([#1014](https://github.com/DevExpress/testcafe-hammerhead/issues/1014))
```
Comment (AlexanderMoskovkin) [inline CHANGELOG.md]: replace please
[#1014](https://github.com/DevExpress/testcafe-hammerhead/issues/1014)
to
[testcafe-hammerhead#1014](https://github.com/DevExpress/testcafe-hammerhead/issues/1014)

like in the v 0.11.0 changes

File: CHANGELOG.md (hunk 7 above)
Comment (helen-dikareva) [inline CHANGELOG.md]: My bad, that's ok. Sorry

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,297 @@
...
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const windowLocation = ClientFunction(() => window.location);
+
+fixture `My Fixture`
+    .page `http://www.example.com`;
+
+test('My Test', asynct t => {
```
Comment (AlexanderMoskovkin) [inline CHANGELOG.md]: a typo: `asynct`

Comment (AlexanderMoskovkin) [inline CHANGELOG.md]: This test doesn't work. Should be
```js
import { ClientFunction } from 'testcafe';

const windowLocation = ClientFunction(() => window.location.toString());

fixture `My Fixture`
    .page `http://www.example.com`;

test('My Test', async t => {
    await t.expect(windowLocation()).eql('http://www.example.com');
});
```

**Pull request merged**